File: leet_code/javascript_30/python/2704.py
class Expect:
    def __init__(self, val):
        self.val = val

    def to_be(self, lav):
        if self.val != lav:
            raise ValueError('Not Equal')
        return True

    def not_to_be(self, lav):
        if self.val == lav:
            raise ValueError('Equal')
        return True

def expect(val):
    return Expect(val)
expect(5).to_be(5)
expect(5).not_to_be(6)
expect(5).to_be(6)

class Expect2:
    def __init__(self, val):
        self.val = val

    def to_be(self, lav):
        if self.val != lav:
            raise ValueError('Not Equal')
        return True

    def not_to_be(self, lav):
        if self.val == lav:
            raise ValueError('Equal')
        return True

def expect2(val):
    return Expect2(val)

expect2(5).to_be(5)
expect2(5).not_to_be(6)
expect2(5).to_be(6)

File: leet_code/javascript_30/go/2704.go
package main

import (
	"errors"
	"fmt"
)

type Expect struct {
	val interface{}
}

func (e *Expect) ToBe(lav interface{}) error {
	if e.val != lav {
		return errors.New("Not Equal")
	}
	return nil
}

func (e *Expect) NotToBe(lav interface{}) error {
	if e.val == lav {
		return errors.New("Equal")
	}
	return nil
}

func expect(val interface{}) *Expect {
	return &Expect{val: val}
}

func main() {
	e := expect(5)
	if err := e.ToBe(5); err != nil {
		fmt.Println(err)
	} else {
		fmt.Println("Equal")
	}

	if err := e.NotToBe(6); err != nil {
		fmt.Println(err)
	} else {
		fmt.Println("Not Equal")
	}
}
